sql/log_event.cc

```diff
@@ -1409,7 +1409,14 @@ code_name(int code)
   case Q_MASTER_DATA_WRITTEN_CODE: return "Q_MASTER_DATA_WRITTEN_CODE";
   case Q_HRNOW: return "Q_HRNOW";
   }
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
notably never reached as `CHECK_SPACE`, the only caller of `code_name` always has a valid code. Changing to a snprint would be less verbose.

Reply:
I made the exact same comment in [MDEV-35299](https://jira.mariadb.org/browse/MDEV-35299) a couple of days ago. Basically, after a proper fix, `git grep -w sprintf` should not return anything.

Reply:
the latest commit doesn't contain any #pragma and ifdef __clang__